src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -238,6 +238,12 @@ static constexpr Property<uint32_t, PropertyMutability::RO> optimal_number_of_in
  */
 static constexpr Property<element::Type, PropertyMutability::RW> inference_precision{"INFERENCE_PRECISION_HINT"};
 
+/**
+ * @brief Core supported property keys
+ * @ingroup ov_runtime_cpp_prop_api
+ */
+static constexpr Property<std::set<std::string>, PropertyMutability::RO> core_property_keys{"CORE_PROPERTY_KEYS"};
```

Comment:
Should we use the same structures like for supported properties?

Reply:
std::set support count() method which is useful to check whether it contains a key, but std::vector doesn't have.